graphql/api/src/main/java/com/expediagroup/streamplatform/streamregistry/graphql/filters/ConsumerBindingFilter.java

```diff
@@ -36,26 +37,31 @@ public ConsumerBindingFilter(ConsumerBindingKeyQuery keyQuery, SpecificationQuer
 
   @Override
   public boolean test(ConsumerBinding d) {
+    return matchesConsumerBindingKey(d.getKey(), keyQuery)
+      && matchesSpecification(d.getSpecification(), specQuery);
+  }
+
+  public static Boolean matchesConsumerBindingKey(ConsumerBindingKey key, ConsumerBindingKeyQuery keyQuery) {
     if (keyQuery != null) {
-      if (!matches(d.getKey().getInfrastructureName(), keyQuery.getInfrastructureNameRegex())) {
+      if (!matches(key.getInfrastructureName(), keyQuery.getInfrastructureNameRegex())) {
         return false;
       }
-      if (!matches(d.getKey().getInfrastructureZone(), keyQuery.getInfrastructureZoneRegex())) {
+      if (!matches(key.getInfrastructureZone(), keyQuery.getInfrastructureZoneRegex())) {
         return false;
       }
-      if (!matches(d.getKey().getStreamDomain(), keyQuery.getStreamDomainRegex())) {
+      if (!matches(key.getStreamDomain(), keyQuery.getStreamDomainRegex())) {
         return false;
       }
-      if (!matches(d.getKey().getStreamName(), keyQuery.getStreamNameRegex())) {
+      if (!matches(key.getStreamName(), keyQuery.getStreamNameRegex())) {
         return false;
       }
-      if (keyQuery.getStreamVersion() != null && d.getKey().getStreamVersion() != keyQuery.getStreamVersion()) {
+      if (keyQuery.getStreamVersion() != null && key.getStreamVersion() != keyQuery.getStreamVersion()) {
```

Comment:
This was there before hand, but we should us `equals` when comparing `Integer`, not `!=`/`==`